pkg/operator/configobservation/images/observe_images.go

```diff
@@ -41,14 +34,19 @@ func ObserveInternalRegistryHostname(genericListers configobserver.Listers, reco
 		return observedConfig, errs
 	}
 	if err != nil {
-		return prevObservedConfig, errs
+		return existingConfig, append(errs, err)
 	}
 
 	internalRegistryHostName := configImage.Status.InternalRegistryHostname
 	if len(internalRegistryHostName) > 0 {
 		if err := unstructured.SetNestedField(observedConfig, internalRegistryHostName, internalRegistryHostnamePath...); err != nil {
 			errs = append(errs, err)
```

Comment:
in what state is `observedConfig` at this point ?

Reply:
from what I've seen in cases like that we return `existingConfig`

Reply:
it's empty. I.e. this will never fail.